src/compute-client/src/protocol/response.rs

```diff
@@ -102,11 +102,10 @@ pub enum ComputeResponse<T = mz_repr::Timestamp> {
     ///
     /// For each subscribe that was installed by a previous [`CreateDataflows` command], the
     /// replica must emit [`Batch`] responses that cover the entire time interval from the
-    /// subscribe dataflow's `as_of` until the subscribe advances to the empty frontier or is
+    /// minimum time until the subscribe advances to the empty frontier or is
     /// dropped. The time intervals of consecutive [`Batch`]es must be increasing, contiguous,
     /// non-overlapping, and non-empty. All updates transmitted in a batch must have times within
-    /// that batch’s time interval. The `upper` of the first [`Batch`] of a subscribe must not be
-    /// less than that subscribe's initial `as_of` frontier.
+    /// that batch’s time interval and must be greater than or equal to `as_of`.
```

Comment:
This drops the requirement about the batch upper, which we absolutely still need! Also, less importantly, this reads as "all updates [...] must be greater than or equal to as_of", which is a confusing wording.